### PR TITLE
NICPS-19 : Search and Delete

### PR DESCRIPTION
**Pull request by chinmay15**
Problem : Provide support for search and delete items in all users.

Fix :
1) Existing perl script available at http://zre-matrix.eng.zimbra.com:8080/@md=d&cd=/depot/psteam/main/tools/&c=79S@/depot/psteam/main/tools/zmsearchNdelete/?ac=83 is added.
2) usage info and details are given in script itself

Testing done : Tested script with single and multi-node environment.

Testing needed by QA : Extensive testing is needed on single and multi-node environment with different type of search criteria.

Linked PR - https://github.com/Zimbra/zm-build/pull/79

### CODE REVIEW

**Comment by plobbes** [comment]
(I left the same comments on https://github.com/Zimbra/zm-build/pull/79)
If we're "productizing" this utility, I think we may want to consider not using mixed case in the utility name
* zmsearchNdelete → zmsearchndelete

If we do that we should update the documentation in the utility. We could also:
1. update the license block
2. change all uses of '&' in the documentation with the word 'and'
3. Rewrite/reword the BUGS/KNOWN issues section. It doesn't flow very well and could use some cleanup.

Up to you if you choose to do any of these things though.

**Comment by chinmay15** [comment]
updated. please review.

File: src/libexec/zmsearchndelete [hunk 1]
```diff
@@ -0,0 +1,946 @@
...
+
+=head1 SYNOPSIS
+
+zmsearchndelete [options] [<userlistfile> ...]
+
+  Options: [*] == Required, [+] == Multiple vals OK, (val) == Default
+    --url <url>            [*]URL of soap service
+    --authuser <user>      [*]account for authentication
+    --password <password>  [*]password for authuser
+    --query 		   [*]search string specified in "" boundary
```
Comment (plobbes) [inline src/libexec/zmsearchndelete]: nit: there are a few tabs embedded in this file (this line being the first) - please replace all tabs with spaces and align properly

File: src/libexec/zmsearchndelete [hunk 2]
```diff
@@ -0,0 +1,946 @@
...
+      :                    undef;
+
+    my $soap = $opt{soap};
+    my $req  = SOAP::Data->name( $opt{req} );
+    $req->attr( $opt{attr} ) if $opt{attr};
+
+    my $resp;
+    my ( $try, $maxtry, $sleep ) = ( 0, 3, 0 );
+    while ( !$resp && $try++ < $maxtry ) {
+		$soap->{_transport}->{_proxy}->{ssl_opts}->{verify_hostname} = 0;
```
Comment (plobbes) [inline src/libexec/zmsearchndelete]: nit: tabs to spaces (and align properly)
fix: documenting that we're not ensuring that servers that have a valid certificate matching the expected hostname

**Comment by chinmay15** [comment]
Snickers branch was deleted and re-created, so had to generate new pull request.
New PR link - https://github.com/Zimbra/zm-core-utils/pull/11

File: src/libexec/zmsearchndelete (hunk 1 above)
Comment (chinmay15) [reply src/libexec/zmsearchndelete]: done

File: src/libexec/zmsearchndelete (hunk 2 above)
Comment (chinmay15) [reply src/libexec/zmsearchndelete]: nit - done
fix- removed bypassing ssls_opts, so that ssl check will be done.

**Pull request closed without merging**